### DISCUSSION TITLE
All clients have the same IP in pihole

### DISCUSSION THREAD

**Discussion by goremykin** [wg-easy/wg-easy#1028] [Q&A]
**Describe the bug**
If I have a few clients all of them have the same IP in pihole and merged into a single row. Is it possible to make it work the same way as pivpn works - each client has it's own IP?
The only information I found is this post:
https://www.reddit.com/r/WireGuard/comments/1ahb2og/comment/koreyel/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button

If I understand it correctly this has something to do with MASQUERADE.
But unfortunately, when I specify the same WG_POST_UP as the post suggests, the VPN simply stops working.

**To to Reproduce**
1. Start wg-easy and pihole containers according to [wiki](https://github.com/wg-easy/wg-easy/wiki/Using-WireGuard-Easy-with-Pi-Hole)
2. Create a few clients
3. Connect to each client
4. Check pihole dashboard

**Expected behavior**
Each client has it's own IP and row

**Screenshots**
![image](https://github.com/wg-easy/wg-easy/assets/25577658/17ee609c-84b2-4cd0-a94f-4baf64776f18)

**Desktop (please complete the following information):**
 - Debian 12 for arm
 - Browser Firefox

**Comment by eandersons** [comment]
Similar question has already been asked before: #94 (without a solution though).

You could try:

- #1016;
- run both Pi-hole and WireGuard Easy on the host network.

**Reply by peterlewis** [reply]
~Alternatively if you run the containers on a macvlan docker network then this, too, can enable client identification in Pi-Hole. This is what I do and it works a treat.~

Please disregard. I am an idiot. Generally, I run my DNS service without any logging enabled (so current state not immediately visible), and was misremembering from when I was testing the inbuilt UniFi WG solution!

**Reply by goremykin** [reply]
@eandersons, Unfortunately, this solution results in no internet (the same problem is described in the comments)

**Comment by goremykin** [comment]
@peterlewis, Could you share your docker-compose or docker run arguments? :)

**Comment by comebackworks** [comment]
Here is how I have solved this.
My pihole and wg-easy container sit on a docker bridge network 10.8.1.0/24, with address 10.8.1.3 and 10.8.1.2 respectively
My wg clients are in the network 10.8.0.0/24
1) Prepend the following in the PostUp Hook (leaving the rest of the line unchanged):
`iptables -t nat -I POSTROUTING 1 -s {{ipv4Cidr}} -d 10.8.1.0/24 -j RETURN;`
2) Prepend the following in the PostDown Hook:
`iptables -t nat -D POSTROUTING -s {{ipv4Cidr}} -d 10.8.1.0/24 -j RETURN;`
In this way, the DNS queries are not NAT'd and leave the wg-easy container with their 10.8.0.x source address.
3) the pi-hole container needs to be able to route the traffic back to the wg container.
Therefore add the ip route inside the pi-hole container.
In docker compose, for the pi-hole service add the following lines:
```yaml
cap_add:
 - NET_ADMIN
entrypoint: ["/bin/sh","-c", "ip route replace 10.8.0.0/24 via 10.8.1.2 dev eth0 || true; exec start.sh" ]
```
